core/src/banking_stage/consumer.rs

```diff
@@ -429,11 +429,19 @@ impl Consumer {
             ..
         } = execute_and_commit_transactions_output;
 
-        QosService::update_or_remove_transaction_costs(
-            transaction_qos_cost_results.iter(),
-            commit_transactions_result.as_ref().ok(),
-            bank,
-        );
+        // once feature `apply_cost_tracker_during_replay` is activated, leader shall no longer
+        // adjust block with executed cost (a behavior more inline with bankless leader), instead
+        // will be exclusively using requested `compute_unit_limit` in cost tracking.
```

Comment:
🤔 
> exclusively using requested `compute_unit_limit` in cost tracking

Sorry to nit-pick this comment, but I want to be really clear about the expectations and behaviors, since this kind of implies that we will require the request to be set (to me at least), and I don't beleive that is the case.

1. We still have a default CU-limit which is used if not specified.
2. If we have pure builtin txs, even if the request is set, we don't use that limit in the cost tracking, only fee calculations.

Are these 2 points still going to be true moving forward?

Might be better to just be a bit more vague in the comments, w/ details being in the cost model itself. Something along the lines of "instead calculated, requested, or default costs will be used in cost tracking".

Reply:
yes for point 1, default will be used if `compute_unit_limit` is not explicitly set. I implicitly included default when mentioning requested cu. But good point, #31700 clarifies it. 

Point 2 is also correct, but it probably should change. Since all builtins now (in 1.16+) also consume requested CUs, probably no need to distinguish builtin and bpf programs in cost model.

Reply:
> Since all builtins now (in 1.16+) also consume requested CUs, probably no need to distinguish builtin and bpf programs in cost model.

I suppose the difference is we know the static builtin costs up-front, compared to bpf. So we can accurately determine those costs, which are (if tx will succeed) strictly <= requested.

Actually, taking it a slight step further - if we found it to be a problem we could immediately reject (try taking fees only) if a builtin-only tx requested too few CUs.